packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/add-node/add-node.routing.js

```diff
@@ -13,8 +13,25 @@ export default /* @ngInject */ ($stateProvider) => {
       layout: 'modal',
       resolve: {
         breadcrumb: () => null,
-        price: /* @ngInject */ (getCurrentFlavor) =>
-          getCurrentFlavor().nodeHourlyPrice,
+        price: /* @ngInject */ (getCurrentFlavor, engines, database) => {
+          const currentFlavor = getCurrentFlavor();
+          const flavorPrice = currentFlavor.nodeHourlyPrice;
+          const dbEngine = engines.find(
+            (engine) => engine.name === database.engine,
+          );
+          let addedStorageByNode =
+            database.disk.size - currentFlavor.minDiskSize;
+          // if engine is distributed, the stockage is distributed between the nodes, so to get the
+          // size of for one node, we need to divide by the number of nodes
+          if (dbEngine.isDistributedStorage) {
+            addedStorageByNode /= database.nodeNumber;
+          }
+          const additionalStoragePrice = Math.max(
+            0,
+            addedStorageByNode * currentFlavor.hourlyPricePerGB.priceInUcents,
+          );
+          return flavorPrice.priceInUcents + additionalStoragePrice;
+        },
```

Comment:
**nitpick:** No need to block the state transition for this calculation. Better to perform this in ctrl.

Reply:
Moved to the controller: https://github.com/ovh/manager/pull/8876/commits/081142295e998fee5b78dcc16e71b6c5b415e990